src/axom/core/tests/core_Path.hpp

```diff
@@ -116,15 +188,25 @@ TEST(core_Path, dirname_basename_multi_component_alt_delim)
   EXPECT_EQ(path.baseName(), "baz");
 }
 
+TEST(core_Path, dirname_basename_multi_component_lead_delim)
+{
+  axom::Path path("/foo/bar/baz");
+  EXPECT_EQ(path.dirName(), "/foo/bar");
+  EXPECT_EQ(path.baseName(), "baz");
+}
+
 TEST(core_Path, basic_equals)
 {
   axom::Path first("foo/bar/baz");
   axom::Path second("foo/bar/baz");
   axom::Path third("foo:bar:baz", ':');
```

Comment:
Suggestion: We should probably also test the leading delimeter with a non-default delim, 
e.g. `Path {":foo:bar:baz", ':' }`